lib/catalog/data_driven_form_validator.rb

```diff
@@ -0,0 +1,86 @@
+module Catalog
+  class DataDrivenFormValidator
+    COMPONENTS = %i[text-field
+                    textarea-field
+                    field-array
+                    select-field
+                    checkbox
+                    sub-form
+                    radio
+                    tabs
+                    tab-item
+                    date-picker
+                    time-picker
+                    wizard
+                    switch-field
+                    plain-text].freeze
+
+    VALIDATORS = %i[required-validator
+                    min-length-validator
+                    max-length-validator
+                    exact-length-validator
+                    min-items-validator
+                    min-number-value
+                    max-number-value
+                    pattern-validator
+                    url-validator].freeze
+
+    DATA_TYPES = %i[integer
+                    float
+                    number
+                    boolean
+                    string].freeze
+
+    def self.valid?(ddf)
+      ddf = ddf.class == Hash ? ddf.with_indifferent_access : JSON.parse(ddf).with_indifferent_access
+      fields?(ddf[:schema][:fields])
+
+      true
+    end
+
+    class << self
+      def fields?(fields)
+        fields.each do |field|
+          component?(field[:component].to_sym)
```

Comment:
@lindgrenj6 Do we need to call the method component? since we are not using its return to check against a boolean value. Should the methods be just called
``
check_component
check_data_type
check_validators
```

Reply:
Updated the method names.